pkg/driver/driver_aws.go

```diff
@@ -115,30 +120,149 @@ func (d *AWSDriver) Create() (string, string, error) {
 		networkInterfaceSpecs = append(networkInterfaceSpecs, spec)
 	}
 
-	// Specify the details of the machine
-	inputConfig := ec2.RunInstancesInput{
-		ImageId:           aws.String(d.AWSMachineClass.Spec.AMI),
-		InstanceType:      aws.String(d.AWSMachineClass.Spec.MachineType),
-		MinCount:          aws.Int64(1),
-		MaxCount:          aws.Int64(1),
-		UserData:          &UserDataEnc,
-		KeyName:           aws.String(d.AWSMachineClass.Spec.KeyName),
-		NetworkInterfaces: networkInterfaceSpecs,
-		IamInstanceProfile: &ec2.IamInstanceProfileSpecification{
-			Name: &(d.AWSMachineClass.Spec.IAM.Name),
-		},
-		BlockDeviceMappings: blkDeviceMappings,
-		TagSpecifications:   []*ec2.TagSpecification{tagInstance, tagVolume},
-	}
+	var instance *ec2.Instance
+	if d.AWSMachineClass.Spec.SpotPrice == nil {
+		// Specify the details of the machine
+		inputConfig := ec2.RunInstancesInput{
+			ImageId:           aws.String(d.AWSMachineClass.Spec.AMI),
+			InstanceType:      aws.String(d.AWSMachineClass.Spec.MachineType),
+			MinCount:          aws.Int64(1),
+			MaxCount:          aws.Int64(1),
+			UserData:          &UserDataEnc,
+			KeyName:           aws.String(d.AWSMachineClass.Spec.KeyName),
+			NetworkInterfaces: networkInterfaceSpecs,
+			IamInstanceProfile: &ec2.IamInstanceProfileSpecification{
+				Name: &(d.AWSMachineClass.Spec.IAM.Name),
+			},
+			BlockDeviceMappings: blkDeviceMappings,
+			TagSpecifications:   []*ec2.TagSpecification{tagInstance, tagVolume},
+		}
 
-	runResult, err := svc.RunInstances(&inputConfig)
-	if err != nil {
-		metrics.APIFailedRequestCount.With(prometheus.Labels{"provider": "aws", "service": "ecs"}).Inc()
-		return "Error", "Error", err
+		runResult, err := svc.RunInstances(&inputConfig)
+		if err != nil {
+			metrics.APIFailedRequestCount.With(prometheus.Labels{"provider": "aws", "service": "ecs"}).Inc()
```

Comment:
I'm not super familiar with this codebase so apologies if this is completely wrong, but is the service supposed to be "ec2" or is "ecs" correct here?

Reply:
You are most certainly right, but this is out of the scope of this PR.